src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -297,12 +296,7 @@ void ComponentFiller::addVariables(const LinearProblemApi::FillContext& ctx)
     const auto& variables = component_.getModel()->Variables();
     auto& pb = optimEntityContainer_.Problem();
 
-    // Skip the variable in case of location mismatch
-    const auto locationFilter = std::views::filter(
-      [&](const auto& variable)
-      { return AreLocationsCompatible(variable.location(), targetLocation_); });
-
-    for (const auto& variable: variables | locationFilter)
+    for (const auto& variable: variables | locationFilter())
```

Comment:
Definition of `locationFilter()` better indeed ! 
Even better : initialise a **locationFilter_** at construction of the **ComponentFiller**.
This would avoid rebuilding the view at each call of `locationFilter()`.
So we'd have : 
```c++
auto locationFilter()
{
    return locationFilter_
}
```